test/dynamo/test_modules.py

```diff
@@ -1330,11 +1333,61 @@ def __torch_function__(cls, func, types, args=(), kwargs=None):
 
                 self.assertEqual(cnt.op_count, 4)
                 self.assertTrue(torch._dynamo.testing.same(out1, out2))
-            finally:
-                torch._dynamo.config.traceable_tensor_subclasses.remove(TensorProxy)
 
         run()
 
+    def test_torch_mangled_class_name(self):
+        original = TensorWithTFOverrideVariable.global_mangled_class_name
+        results = []
+
+        def instrumented(self, tx):
+            result = original(self, tx)
+            results.append(result)
+            return result
+
+        TensorWithTFOverrideVariable.global_mangled_class_name = instrumented
+
+        def one_break(x):
+            x = F.sigmoid(x)
+            print()  # force break
+            x = x.sigmoid()
+            return x
+
+        try:
+            with temporary_tensor_subclass() as TensorProxy:
+                x = torch.randn(1).as_subclass(TensorProxy)
+                x1 = one_break(x)
+
+                cnt = torch._dynamo.testing.CompileCounter()
+                opt_one_break = torch._dynamo.optimize(cnt)(one_break)
+                x2 = opt_one_break(x)
+
+                self.assertTrue(torch._dynamo.testing.same(x1, x2))
+                self.assertEqual(cnt.frame_count, 2)
+                self.assertEqual(cnt.op_count, 2)
+
+                compile_ids = set()
+                for r in results:
+                    # A mangled classname looks like __subclass_TensorProxy_94524181138240_c0
+                    # where the last segment contains the compile_id.
+                    prefix = "__subclass_TensorProxy_"
+                    before, sep, after = r.partition(prefix)
+                    self.assertEqual(before, "")
+                    self.assertEqual(sep, prefix)
+
+                    class_type_id, compile_id = after.split("_")
+                    self.assertTrue(class_type_id.isnumeric())
+                    self.assertTrue(compile_id.startswith("c"))
+
+                    cid = compile_id[1:]
+                    self.assertTrue(cid.isnumeric())
+                    compile_ids.add(int(cid))
+
+                self.assertEqual(compile_ids, set(range(3)))
```

Comment:
We should just check that they're unique, not the exact value. The exact value might be flaky

Reply:
Very good point, I just collected the compile ids as strings into a set, and checked that the size of the set was 3.

Reply:
Hah, right after you said this, one instance of this unit test failed, showing that this wasn't hypothetical. :-D Let's see how the CI shakes out, but the last change did seem to fix that test locally.